src/TuDa.CIMS.Api/Repositories/PurchaseRepository.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Microsoft.EntityFrameworkCore;
+using TuDa.CIMS.Api.Database;
+using TuDa.CIMS.Api.Interfaces;
+using TuDa.CIMS.Shared.Dtos;
+using TuDa.CIMS.Shared.Entities;
+
+namespace TuDa.CIMS.Api.Repositories;
+
+public class PurchaseRepository : IPurchaseRepository
+{
+    private readonly CIMSDbContext _context;
+
+    public PurchaseRepository(CIMSDbContext context)
+    {
+        _context = context;
+    }
+
+    /// <summary>
+    /// Returns all purchases from the database.
+    /// </summary>
+    /// <param name="workingGroupId">the specific ID of the workinggroup</param>
+    /// <returns></returns>
+    public async Task<IEnumerable<Purchase>> GetAllAsync(Guid workingGroupId)
+    {
+        return await _context.WorkingGroups
+            .Where(i => i.Id == workingGroupId)
+            .SelectMany(i => i.Purchases)
+            .Include(i => i.Buyer)
+            .Include(i => i.Entries)
+            .ToListAsync();
+    }
+
+    /// <summary>
+    /// Returns a single purchase by its specific ID.
+    /// </summary>
+    /// <param name="id">the specific ID for the purchase</param>
+    /// <param name="workingGroupId">the specific id of a workinggroup</param>
+    /// <returns></returns>
+    public async Task<Purchase?> GetOneAsync(Guid id, Guid workingGroupId)
+    {
+        return await _context.WorkingGroups
+            .Where(i => i.Id == workingGroupId)
+            .SelectMany(i => i.Purchases)
+            .Where(i => i.Id == id)
+            .Include(i => i.Buyer)
+            .Include(i => i.Entries)
+            .SingleOrDefaultAsync();
+    }
+
+    /// <summary>
+    /// Updates an existing purchase by its ID using the provided update model.
+    /// </summary>
+    /// <param name="id">the specific ID of the purchase</param>
+    /// <param name="workingGroupId">the specific id of a workinggroup</param>
+    /// <param name="updateModel">the model containing the update values for the purchase </param>
+    /// <returns></returns>
+    public async Task<ErrorOr<Updated>> UpdateAsync(Guid id, Guid workingGroupId, UpdatePurchaseDto updateModel)
+    {
+        var existingItem = await _context
+            .WorkingGroups
+            .Where(i => i.Id == workingGroupId)
+            .SelectMany(i => i.Purchases)
+            .Where(i => i.Id == id)
+            .Include(i => i.Buyer)
+            .Include(i => i.Entries)
+            .SingleOrDefaultAsync();
+
+        if (existingItem is null)
+        {
+            return Error.NotFound("Purchase.update", $"Purchase with ID {id} was not found.");
+        }
+
+
+        if (updateModel.Buyer is not null)
+        {
+            var buyer = await _context.Persons.SingleOrDefaultAsync(r => r.Id == updateModel.Buyer.Id);
```

Comment:
🔧 🤔 When only the ID is needed of the `Buyer` the `UpdateDto` should only have a field for that